daal4py/sklearn/ensemble/_forest_0_23.py

```diff
@@ -0,0 +1,744 @@
+                      FutureWarning)
+
+        if self.min_impurity_split < 0.:
+            raise ValueError("min_impurity_split must be greater than "
+                             "or equal to 0")
+    if self.min_impurity_decrease < 0.:
+        raise ValueError("min_impurity_decrease must be greater than "
+                         "or equal to 0")
+    if self.max_leaf_nodes is not None:
+        if not isinstance(self.max_leaf_nodes, numbers.Integral):
+            raise ValueError("max_leaf_nodes must be integral number but was "
+                             "%r" % self.max_leaf_nodes)
+        if self.max_leaf_nodes < 2:
+            raise ValueError(("max_leaf_nodes {0} must be either None "
+                              "or larger than 1").format(self.max_leaf_nodes))
+
+def _daal_fit_classifier(self, X, y, sample_weight=None):
+    y = check_array(y, ensure_2d=False, dtype=None)
+    y, expanded_class_weight = self._validate_y_class_weight(y)
+    n_classes_ = self.n_classes_[0]
+    classes_ = self.classes_[0]
+    self.n_features_ = X.shape[1]
+
+    if expanded_class_weight is not None:
+        if sample_weight is not None:
+            sample_weight = sample_weight * expanded_class_weight
+        else:
+            sample_weight = expanded_class_weight
+    if sample_weight is not None:
+        sample_weight = [sample_weight]
+
+    rs_ = check_random_state(self.random_state)
+    seed_ = rs_.randint(0, np.iinfo('i').max)
+
+    if n_classes_ < 2:
+        raise ValueError("Training data only contain information about one class.")
+
+    # create algorithm
+    X_fptype = getFPType(X)
+    daal_engine_ = daal4py.engines_mt2203(seed=seed_, fptype=X_fptype)
+    _featuresPerNode = _to_absolute_max_features(self.max_features, X.shape[1], is_classification=True)
+
+    n_samples_bootstrap = _get_n_samples_bootstrap(
+        n_samples=X.shape[0],
+        max_samples=self.max_samples
+    )
+
+    if not self.bootstrap and self.oob_score:
+        raise ValueError("Out of bag estimation only available"
+                         " if bootstrap=True")
+
+    dfc_algorithm = daal4py.decision_forest_classification_training(
+        nClasses = int(n_classes_),
+        fptype = X_fptype,
+        method = 'defaultDense',
+        nTrees = int(self.n_estimators),
+        observationsPerTreeFraction = n_samples_bootstrap if self.bootstrap == True else 1.,
+        featuresPerNode = int(_featuresPerNode),
+        maxTreeDepth = int(0 if self.max_depth is None else self.max_depth),
+        minObservationsInLeafNode = (self.min_samples_leaf if isinstance(self.min_samples_leaf, numbers.Integral)
+                                     else int(ceil(self.min_samples_leaf * X.shape[0]))),
+        engine = daal_engine_,
+        impurityThreshold = float(0.0 if self.min_impurity_split is None else self.min_impurity_split),
+        varImportance = "MDI",
+        resultsToCompute = "",

```

Comment:
For parameter `oob_score = True` we should compute with flags: `computeOutOfBagError`, `computeOutOfBagErrorPerObservation` ?

Reply:
We do not have parameters for calculating oob score. We calculated this with the outOfBagError property, but it always returns a NoneType.